tests/migrations/test_operations.py

```diff
@@ -2653,89 +2653,116 @@ def create_data(models, schema_editor):
-            Article.objects.create(id=2 ** 33, name="Django2", blog=blog2)
-
-        create_blog = migrations.CreateModel(
-            "Blog",
-            [
-                ("id", models.AutoField(primary_key=True)),
-                ("name", models.CharField(max_length=100)),
-            ],
-            options={},
-        )
-        create_article = migrations.CreateModel(
-            "Article",
-            [
-                ("id", models.AutoField(primary_key=True)),
-                ("blog", models.ForeignKey(to="test_blog.Blog", on_delete=models.CASCADE)),
-                ("name", models.CharField(max_length=100)),
-                ("data", models.TextField(default="")),
-            ],
-            options={},
-        )
-        fill_initial_data = migrations.RunPython(create_initial_data, create_initial_data)
-        fill_big_data = migrations.RunPython(create_big_data, create_big_data)
+        A field may be migrated in the following ways:
 
-        grow_article_id = migrations.AlterField("Article", "id", models.BigAutoField(primary_key=True))
-        grow_blog_id = migrations.AlterField("Blog", "id", models.BigAutoField(primary_key=True))
+        - AutoField to BigAutoField.
+        - SmallAutoField to AutoField.
+        - SmallAutoField to BigAutoField.
+        """
+        tests = [
+            (models.AutoField, models.BigAutoField, 2 ** 22, 2 ** 33),
+            (models.SmallAutoField, models.AutoField, 2 ** 11, 2 ** 22),
+            (models.SmallAutoField, models.BigAutoField, 2 ** 11, 2 ** 33),
+        ]
 
-        project_state = ProjectState()
-        new_state = project_state.clone()
-        with connection.schema_editor() as editor:
-            create_blog.state_forwards("test_blog", new_state)
-            create_blog.database_forwards("test_blog", editor, project_state, new_state)
+        for source_field, target_field, source_value, target_value in tests:
+            with self.subTest('%s to %s' % (source_field.__name__, target_field.__name__)):
+                def create_initial_data(models, schema_editor):
+                    Article = models.get_model("test_article", "Article")
+                    Blog = models.get_model("test_blog", "Blog")
+                    blog = Blog.objects.create(name="web development done right")
+                    Article.objects.create(name="Frameworks", blog=blog)
+                    Article.objects.create(id=source_value, name="Programming Languages", blog=blog)
+
+                def create_big_data(models, schema_editor):
+                    Article = models.get_model("test_article", "Article")
+                    Blog = models.get_model("test_blog", "Blog")
+                    blog2 = Blog.objects.create(name="Frameworks", id=target_value)
+                    Article.objects.create(name="Django", blog=blog2)
+                    Article.objects.create(id=target_value, name="Django2", blog=blog2)
```

Comment:
I don't think that we need a `source_value`. You can move `create_initial_data()` outside this loop to the previous place. This should reduce the diff.